packages/flutter/lib/src/material/navigation_rail.dart

```diff
@@ -0,0 +1,761 @@
+  final int currentIndex;
+
+  /// Called when one of the [destinations] is selected.
+  ///
+  /// The stateful widget that creates the navigation rail needs to keep
+  /// track of the index of the selected [NavigationRailDestination] and call
+  /// `setState` to rebuild the navigation rail with the new [currentIndex].
+  final ValueChanged<int> onDestinationSelected;
+
+  /// The elevation for the inner side of the rail.
+  ///
+  /// The shadow only shows on the inner side of the rail.
+  ///
+  /// In LTR configurations, the inner side is the right side, and in RTL
+  /// configurations, it is the left side.
+  final double elevation;
+
+  /// The alignment for the [NavigationRailDestination]s as they are positioned
+  /// within the [NavigationRail].
+  ///
+  /// Navigation rail destinations can be aligned as a group to the [top],
+  /// [bottom], or [center] of a layout.
+  final NavigationRailGroupAlignment groupAlignment;
+
+  /// Defines the layout and behavior of the labels for the default, unextended
+  /// [NavigationRail].
+  ///
+  /// When the navigation rail is extended, the labels are always shown.
+  ///
+  /// See also:
+  ///
+  ///   * [NavigationRailLabelType] for information on the meaning of different
+  ///   types.
+  final NavigationRailLabelType labelType;
+
+  /// The [TextStyle] of the unselected [NavigationRailDestination] labels.
+  ///
+  /// When the [NavigationRailDestination] is selected, the
+  /// [selectedLabelTextStyle] will be used instead.
+  final TextStyle unselectedLabelTextStyle;
+
+  /// The [TextStyle] of the [NavigationRailDestination] labels when they are
+  /// selected.
+  ///
+  /// When the [NavigationRailDestination] is not selected,
+  /// [unselectedLabelTextStyle] will be used.
+  final TextStyle selectedLabelTextStyle;
+
+  /// The default size, opacity, and color of the icon in the
+  /// [NavigationRailDestination].
+  ///
+  /// If this field is not provided, or provided with any null properties, then
+  /// a copy of the [IconThemeData.fallback] with a custom [NavigationRail]
+  /// specific color will be used.
+  final IconThemeData unselectedIconTheme;
+
+  /// The size, opacity, and color of the icon in the selected
+  /// [NavigationRailDestination].
+  ///
+  /// When the [NavigationRailDestination] is not selected,
+  /// [unselectedIconTheme] will be used.
+  final IconThemeData selectedIconTheme;
+
+  /// The smallest possible width for the rail regardless of the destination
+  /// content size.
+  ///
+  /// The default is 72.
+  ///
+  /// This value also defines the min width and min height of the destination
+  /// boxes.
+  ///
+  /// To make a compact rail, set this to 56 and use
+  /// [NavigationRailLabelType.none].
+  final double preferredWidth;
+
+  /// The final width when the animation is complete for setting [extended] to
+  /// true.
+  ///
+  /// This is only used when [extended] is set to true.
+  ///
+  /// The default value is 256.
```

Comment:
Updated to minExtendedWidth to match minWidth.

I think this is consistent with the way minWidth works, which would be trickier to work with a user defined padding because it has to work with only icons, and icons with labels.

Reply:
Sounds good